### DISCUSSION TITLE
An Idea About Multicast Delegate

### DISCUSSION THREAD

**Discussion by Wit-2k** [dotnet/csharplang#8511] [Language Ideas] [answered]
Now we can use `+`, `-` and their compound assignment operators to deal with multicast delegate, but I think if we can just use the name of method as the operand? 
### Example
```c#
Action act1 = Class1.methodX;
Action act2 = Class2.methodY;
Action act3 = act1 + act2;
act3 -= act2;
```
The above is legal and there's no problem, but it would be more intuitive if the following code is legal as well.
```c#
Action act = Class1.methodX + Class2.methodY;
```

### Drawback
Since methods are invoked sequentially in a multicast delegate, the new form may cause some problems. 

### Conclusion
It seems that this proposal doesn't matter. However, I'd like to do some work to figure out if I can make it by overloading.

**Comment by FaustVX** [comment, marked as answer]
Just cast the first method to the `delegate` type:

```c#
Action act = (Action)Class1.methodX + Class2.methodY;
```
It's because the compiler can't implies the type from `method group` to `Action`

**Reply by Wit-2k** [reply]
Thanks for help.
